windows/hid.c

```diff
@@ -21,74 +21,48 @@
 ********************************************************/
 
 #if defined(_MSC_VER) && !defined(_CRT_SECURE_NO_WARNINGS)
-// Do not warn about mbsrtowcs and wcsncpy usage.
+// Do not warn about wcsncpy usage.
 // https://docs.microsoft.com/cpp/c-runtime-library/security-features-in-the-crt
 #define _CRT_SECURE_NO_WARNINGS
 #endif
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "hidapi.h"
+
 #include <windows.h>
 
 #ifndef _NTDEF_
 typedef LONG NTSTATUS;
 #endif
 
 #ifdef __MINGW32__
-#include <devpropdef.h>
 #include <ntdef.h>
 #include <winbase.h>
+#define WC_ERR_INVALID_CHARS 0x00000080
 #endif
 
 #ifdef __CYGWIN__
 #include <ntdef.h>
+#include <wctype.h>
 #define _wcsdup wcsdup
 #endif
 
-/* The maximum number of characters that can be passed into the
-   HidD_Get*String() functions without it failing.*/
-#define MAX_STRING_WCHARS 0xFFF
-
 /*#define HIDAPI_USE_DDK*/
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-	#include <setupapi.h>
-	#include <winioctl.h>
-	#ifdef HIDAPI_USE_DDK
-		#include <hidsdi.h>
-	#endif
-
-	/* Copied from inc/ddk/hidclass.h, part of the Windows DDK. */
-	#define HID_OUT_CTL_CODE(id)  \
-		CTL_CODE(FILE_DEVICE_KEYBOARD, (id), METHOD_OUT_DIRECT, FILE_ANY_ACCESS)
-	#define IOCTL_HID_GET_FEATURE                   HID_OUT_CTL_CODE(100)
-	#define IOCTL_HID_GET_INPUT_REPORT              HID_OUT_CTL_CODE(104)
-
-#ifdef __cplusplus
-} /* extern "C" */
-#endif
-
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
```

Comment:
whese are removed, not moved - is that mean we don't use any of `stdlib.h`/`string.h` anymore?

Reply:
it seems. or these are included by some other headers. not sure how to check.

Reply:
> or these are included by some other headers

That is a time bomb. I've had a failed builds after a compiler/library upgrade, because some of the headers where implicitly included where they shouldn't.